### PR TITLE
V3 WIP: Add utils and AggregateTool.

### PR DESCRIPTION
**Pull request by evanweible-wf**

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/dart_dev_runner.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    argParser
+      ..addFlag('verbose',
+          abbr: 'v', negatable: false, help: 'Enables verbose logging.')
+      ..addFlag('version',
+          negatable: false, help: 'Prints the dart_dev version.');
+  }
+
+  @override
+  ArgResults parse(Iterable<String> args) {
+    // TODO: get this completion working with bash/zsh
```
Comment (smaifullerton-wk) [inline lib/src/dart_dev_runner.dart]: TODO

File: lib/src/executable.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+final _log = Logger('DartDev');
+
+Future<int> run(List<String> args) async {
+  attachLoggerToStdio(args);
+
+  if (!File(_customDevDartPath).existsSync()) {
+    final toolDir = p.join(p.absolute(p.current), 'tool');
+    stderr
+      ..writeln('Could not find a file named "dev.dart" in "$toolDir".')
+      ..writeln('More info: https://github.com/Workiva/dart_dev#TODO');
```
Comment (smaifullerton-wk) [inline lib/src/executable.dart]: TODO

File: lib/src/executable.dart [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    config = configGetter();
+  } catch (error) {
+    stderr
+      ..writeln('Invalid "tool/dev.dart" in ${p.absolute(p.current)}')
+      ..writeln()
+      ..writeln('It should provide a `Map<String, DevTool> config;` getter,'
+          ' but it either does not exist or threw unexpectedly:')
+      ..writeln('  $error')
+      ..writeln()
+      ..writeln('For more info: http://github.com/Workiva/dart_dev#TODO');
```
Comment (smaifullerton-wk) [inline lib/src/executable.dart]: TODO

**Pull request merged**